src/bundles/ipfs-provider.js

```diff
@@ -239,7 +255,31 @@ const initIPFS = async (store) => {
   store.dispatch({ type: 'IPFS_INIT_STARTED' })
 
   /** @type {Model} */
-  const { apiAddress } = store.getState().ipfs
+  let { apiAddress } = store.getState().ipfs
+
+  // if a custom JSON config is present, use it instead of multiaddr or URL
```

Comment:
Would be nice to encapsulate this `apiAddress` reading into a single function.  Maybe `asAPIAddress` should become `asAPIOptions(string) => string | { ... } | null`. If I recall correctly `apiAddress` is passed through `JSON.stringify` anyway before it's stored to localStorage.

Reply:
Good idea. I've removed double JSON serialization and renamed `asAPIAddress` to `asAPIOptions` in [`83009bb` (#1613)](https://github.com/ipfs-shipyard/ipfs-webui/pull/1613/commits/83009bbca5728ba2a3a5bd19ceb09c66f830048b)